packages/replay/src/coreHandlers/handleGlobalEvent.ts

```diff
@@ -16,51 +16,54 @@ export function handleGlobalEventListener(
-      return event;
-    }
+      // We only want to handle errors & transactions, nothing else
+      if (!isErrorEvent(event) && !isTransactionEvent(event)) {
+        return event;
+      }
 
-    // Unless `captureExceptions` is enabled, we want to ignore errors coming from rrweb
-    // As there can be a bunch of stuff going wrong in internals there, that we don't want to bubble up to users
-    if (isRrwebError(event, hint) && !replay.getOptions()._experiments.captureExceptions) {
-      __DEBUG_BUILD__ && logger.log('[Replay] Ignoring error from rrweb internals', event);
-      return null;
-    }
+      // Unless `captureExceptions` is enabled, we want to ignore errors coming from rrweb
+      // As there can be a bunch of stuff going wrong in internals there, that we don't want to bubble up to users
+      if (isRrwebError(event, hint) && !replay.getOptions()._experiments.captureExceptions) {
+        __DEBUG_BUILD__ && logger.log('[Replay] Ignoring error from rrweb internals', event);
+        return null;
+      }
 
-    // When in buffer mode, we decide to sample here.
-    // Later, in `handleAfterSendEvent`, if the replayId is set, we know that we sampled
-    // And convert the buffer session to a full session
-    const isErrorEventSampled = shouldSampleForBufferEvent(replay, event);
+      // When in buffer mode, we decide to sample here.
+      // Later, in `handleAfterSendEvent`, if the replayId is set, we know that we sampled
+      // And convert the buffer session to a full session
+      const isErrorEventSampled = shouldSampleForBufferEvent(replay, event);
 
-    // Tag errors if it has been sampled in buffer mode, or if it is session mode
-    // Only tag transactions if in session mode
-    const shouldTagReplayId = isErrorEventSampled || replay.recordingMode === 'session';
+      // Tag errors if it has been sampled in buffer mode, or if it is session mode
+      // Only tag transactions if in session mode
+      const shouldTagReplayId = isErrorEventSampled || replay.recordingMode === 'session';
 
-    if (shouldTagReplayId) {
-      event.tags = { ...event.tags, replayId: replay.getSessionId() };
-    }
+      if (shouldTagReplayId) {
+        event.tags = { ...event.tags, replayId: replay.getSessionId() };
+      }
 
-    // In cases where a custom client is used that does not support the new hooks (yet),
-    // we manually call this hook method here
-    if (afterSendHandler) {
-      // Pretend the error had a 200 response so we always capture it
-      afterSendHandler(event, { statusCode: 200 });
-    }
+      // In cases where a custom client is used that does not support the new hooks (yet),
+      // we manually call this hook method here
+      if (afterSendHandler) {
+        // Pretend the error had a 200 response so we always capture it
+        afterSendHandler(event, { statusCode: 200 });
+      }
 
-    return event;
-  };
+      return event;
+    },
+    { id: 'Replay' },
```

Comment:
What's the id for?

Reply:
this is used to identify which event processor dropped an event (possibly). It's optional, but figured it's good form to add this for all our own event processors!